src/test/java/org/assertj/core/api/iterable/IterableAssert_hasExactlyElementsOfTypes_Test.java

```diff
@@ -0,0 +1,22 @@
+package org.assertj.core.api.iterable;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.util.Lists.newArrayList;
+
+import java.util.List;
+
+import org.assertj.core.internal.ObjectArrays;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Only make one test since assertion is delegated to {@link ObjectArrays} which has its own tests.
+ */
+class IterableAssert_hasExactlyElementsOfTypes_Test
```

Comment:
should inherit from `IterableAssertBaseTest` and implement `invoke_api_method` and `verify_internal_effects`

here we only want to verify that `assertHasExactlyElementsOfTypes` is called with the proper arguments.

Reply:
Hi. I am having problems implementing <code>verify_internal_effects</code>. According to my understanding of <code>IterableAssertBaseTest</code>, it tests a method which delegates its function to the instance <code>iterables</code> of the class <code>Iterables</code>. However, <code>hasExactlyElementsOfTypes</code> does not delegate to <code>iterables</code>. Instead, it delegates to <code>ObjectArrays.instance()</code>, so I think the test cannot implement <code>verify_internal_effects</code>. Could you please tell me what can I do?

Reply:
true, try using bits and pieces from ObjectArraysBaseTest (copy what you need in your test class)